ships/0026-image-abstraction-layer.md

```diff
@@ -0,0 +1,232 @@
+Manifest content's hash, users can then refer to the Container Image by its name – and be sure to
+use that specific version. 
+
+In summary, the proposed operator mirrors and indexes remote Container Images into a Kubernetes
+cluster, allowing for other Operators to easily navigate through these multiple indexed Images.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Summary
+
+Design the Kubernetes API types (CRDs) and create a minimal viable product for such an abstraction
+layer.
+
+### Non-Goals
+
+At this stage to provide the integration with Shipwright’ builds is not a goal of this proposal,
+this will need to be tackled once we have the basic types and functionality in place. To provide
+Image storage for Container Images in the form of an integrated Image Registry inside the cluster
+is also not part of the scope of this proposal.
+
+## Proposal
+
+The operator would leverage a _custom resource definition_ called Image. An Image represents an
+image tag in a remote registry. For instance, an Image called `myapp-devel` may be created to keep
+track of the image `quay.io/company/myapp:latest`. A Image _custom resource_ layout would look
+like this:
+
+```yaml
+apiVersion: shipwright.io/v1alpha1
+kind: Image
+metadata:
+  name: myapp-devel
+  namespace: my-namespace
+spec:
+  from: quay.io/company/myapp:latest
+  mirror: false
+  insecure: false
+```
+The meaning of eachproperty in an `Image.Spec` is as follow:
+
+| Property   | Description                                                                       |
+| ---------- | --------------------------------------------------------------------------------- |
+| from       | Indicates the source of the image (from where to import it)                       |
+| mirror     | Informs if the Image should be mirrored to another registry                       |
+| insecure   | Indicates that we should skip tls verification during the image import/mirror     |
+
+Once such a custom_ resource_ is created (with `kubectl create`, for example) the operator won’t
+act on it immediately but it will instead wait until an ImageImport, referring to the created
+Image object, is instantiated by the user.
+
+An ImageImport object looks like:
+
+```
+apiVersion: shipwright.io/v1alpha1
+kind: ImageImport
+metadata:
+  name: myapp-import-00
+  namespace: my-namespace
+Spec:
+  targetImage: myapp-devel
+  from: quay.io/company/myapp:latest
+  mirror: false
+  insecure: false
+```
+
+All Image properties also show up in an ImageImport object, so, if the user does not provide data
+for these properties in the ImageImport the operator will use whatever is set in the Image instead.
+Follow a list of meaning for each property in an ImageImport.Spec:
```

Comment:
Does that make sense ? Would either `targetImage` or the other three properties be a reasonable alternative?